### ISSUE TITLE
Samples should not have unused parameters

### ISSUE THREAD

**Issue by johnkord** [openenclave/openenclave#2220]
Based on the comments from PR #2204, the samples should not require people to cast parameters to void because they are unused.  The functions should be designed to not take those unused parameters in the first place

**Comment by radhikaj** [comment]
Investigate if we can change sealed_data_size param in edl to a constant. Or use the param and check whether it is what you would expect.

**Comment by radhikaj** [comment]
@ryanhsu19 will investigate.

**Comment by ryanhsu19** [comment]
In most samples, "level" remains unused in debug message printing functions such as my_debug() in samples/attested_tls/client/enc/client.cpp and samples/attested_tls/server/enc/server.cpp. 
However, when it gets to the bottom most related functions require the use of mbedtls_debug_print_msg(), which compares the debug threshold and the level before doing the rest. Therefore, I did not remove it.

**Comment by ryanhsu19** [comment]
Currently I am trying to remove the "data" parameter from cert_verify_callback() because it remains voided and unused in all functions (4 in samples/attested_tls) in the sample/attested_tls. There is no problem compiling them, but it will be stuck with on tls_e2e with an error OE_PLATFORM_ERROR returned during ctest, and I am still trying to figure out why.

**Comment by ryanhsu19** [comment]
Also, Alvin and I found that, by using oegdb to debug tests/tls_e2e/host/host.cpp, a signal SIGILL (illegal instruction) returns whenever it tries to run a server. By laying out the ASM, it seems to happen when it executes cpuid in mbedtls_aesni_has_support() in library/aesni.c. It also happened before I started modifying any file. Any ideas?

**Comment by johnkord** [comment]
Hey @ryanhsu19 that's expected! The CPUID instruction is considered illegal inside an SGX enclave.  https://software.intel.com/en-us/node/703005
There's a "vectored exception handler" framework in Open Enclave that properly handles this situation and returns the CPUID values. In the end, CPUID ends up working through an OCALL. However from a debugger, you'll be notified that the illegal instruction signal was sent. You can ignore this signal inside GDB if you want by running `ignore SIGILL`

**Comment by ryanhsu19** [comment]
@johnkord Got it, thanks! I will try again with the condition soon.

**Comment by radhikaj** [comment]
@ryanhsu19 
In this function:
int ecall_dispatcher::unseal_data(
    sealed_data_t* sealed_data,
    size_t sealed_data_size,
    unsigned char** data,
    size_t* data_size)

We dont use sealed_data_size and in fact need to  (void)sealed_data_size; to ensure that the compiler does not throw an unused variable error.

I believe that sealed_data_size is always IV_SIZE, but that needs to be double checked.
if there is a way to specify a constant array dimension in edl, we should prefer that since the size of the array being passed in does not vary across calls.

**Comment by ryanhsu19** [comment]
So far what I have tried is, if I change [in, size = sealed_data_size] to [in, size = 16], ctest will pass but not if I replace it with [in, size = IV_SIZE]. I consider it a macro but it somehow treats it as a variable/attribute and replace it with the corresponding number afterwards.
When I am trying these conditions I have already replaced the variable sealed_data_size instead of the macro IV_SIZE in ecall_dispatcher::unseal_data.
I will continue to consult the SDK Reference and try more.
